### ISSUE TITLE
pofilter -tprintf problems

### ISSUE THREAD

**Issue by asyschikov** [translate/translate#3374]
1) printf test passes this pair:

``` po
msgid "Your question % s"
msgstr "Jouw vraag"
```

While "% s" is actually a valid printf formatter, example: https://ideone.com/AZxWh5
From specification: https://en.wikipedia.org/wiki/Printf_format_string#Flags_field 

2) printf passes this pair

``` po
msgid "Your question %"
msgstr "Jouw vraag"
```

Which will break python formatting, Python doesn't allow single % (while it is valid printf). Literal % should be screened as %% in Python. This is not strictly related to printf because it is valid printf but there is no test to validate it.

**Comment by asyschikov** [comment]
2) I looked more into it, it is not only python issue. Literal % chars must be screened as %% in C as well.

**Comment by asyschikov** [comment]
Basically [this regex](https://github.com/translate/translate/blob/61782aa8fcc31bda13ee4897ec2ef4887b5d45e8/translate/filters/checks.py#L54) should include space in flags section.

**Comment by unho** [comment]
I don't think this is a critical failure since IMHO it is a good practice to place "%s" together and not separate like "% s". But if you are willing to contribute a PR with the appropriate changes and append some tests for it in https://github.com/translate/translate/blob/61782aa8fcc31bda13ee4897ec2ef4887b5d45e8/translate/filters/test_checks.py#L584-L636 we would review it.

**Comment by asyschikov** [comment]
@unho it is not good practice but it is critical failure, which I recently witnessed. The text was `Value is %(some_var)s%% something`, translator translated into another language (I will use English) as `Value is %(some_var)s% something`. It passed print_f test, but in this piece there are 2 parameters introduced, and new one is `% s`. And this resulted in error because there was no value provided for second parameter. I looked into print_f standard deeper and it doesn't allow any % signs that are not 1) part of *valid* parameter definition 2) part of %% literal. Self standing % sign is invalid. I will write a PR.

**Comment by asyschikov** [comment]
@unho https://github.com/translate/translate/pull/3375

**Comment by unho** [comment]
As commented in https://github.com/translate/translate/pull/3375#issuecomment-209298996 I am postponing it since the proposed fix introduces a 20% increase in the time it takes to run the printf check. After 2.0 release we can revisit this.
